### PR TITLE
TermSet Update #1

### PR DESCRIPTION
**Pull request by mavaylon1**
## Motivation

What was the reasoning behind this change? Please explain the changes briefly.
This is the first stage in the TermSet updates. Update `add_ref_term_set` so that it takes in only a root container, in PyNWB it will be a NWBFile, and search for objects that have a `TermSet`. Currently, it is only data objects that can have termsets.

TODO Items:

- [x] Update add_ref_term_set
- [x] Updates test_resources.py
- [x] Update container.py to support `all_children` search
- [x] Update test_container.py to test `all_children`
- [x] Removed `add_ref_termset_ from the gallery for HERD to be updated once TermSet is more finalized with the wrapper.

Changes to Note:
I removed the validation step in `add_ref_termset` (and corresponding tests). This is due to the fact that when a user adds data to `HERD` the data is validated prior using the termset and so we do not need a double validation.

## How to test the behavior?
```
Show how to reproduce the new behavior (can be a bug fix or a new feature)
```
Run unit tests.
## Checklist

- [x] Did you update CHANGELOG.md with your changes?
- [x] Have you checked our [Contributing](https://github.com/hdmf-dev/hdmf/blob/dev/docs/CONTRIBUTING.rst) document?
- [x] Have you ensured the PR clearly describes the problem and the solution?
- [x] Is your contribution compliant with our coding style? This can be checked running `ruff` from the source directory.
- [x] Have you checked to ensure that there aren't other open [Pull Requests](https://github.com/hdmf-dev/hdmf/pulls) for the same change?
- [x] Have you included the relevant issue number using "Fix #XXX" notation where XXX is the issue number? By including "Fix #XXX" you allow GitHub to close issue #XXX when the PR is merged.

### CODE REVIEW

**Comment by mavaylon1** [comment]
Fix #927

**Comment by mavaylon1** [comment]
@oruebel Proof of concept is done but some questions are floating in my head.
1. I need to think about how this all works with compound data and TermSets
2. Looking at the plot_external_resources.py I have to set the parent but I wonder if that's not the most user friendly line to have there. I technically could just use the table as a root_container. Thoughts? (I do need to re-word that section)

Update:
1. We discussed and made a ticket #938 
2. I removed it per consulting with Ryan

File: src/hdmf/container.py [hunk 1]
```diff
@@ -302,6 +303,37 @@ def get_ancestor(self, **kwargs):
...
+            else: # pragma: no cover
+                # warn that a child does not have an object_id, which is unusual
+                warn('%s "%s" does not have an object_id' % (type(n).__class__, n.name))
+            if hasattr(n, 'children'):
+                for c in n.children:
+                    stack.append(c)
+        return ret
+
+    @property
+    def all_objects(self):
```
Comment (rly) [inline src/hdmf/container.py:331]:
```suggestion
    def all_objects(self):
        """Get a LabelledDict that indexed all child objects recursively by object ID."""
```

File: src/hdmf/container.py [hunk 2]
```diff
@@ -302,6 +303,38 @@ def get_ancestor(self, **kwargs):
             p = p.parent
         return None
 
+    def all_children(self):
+        """
+        This method will search through all children within the object and their children.
```
Comment (rly) [inline src/hdmf/container.py]:
```suggestion
        """Get a list of all child objects and their child objects recursively.
```

Comment (rly) [reply src/hdmf/container.py]: In general, the first line of a method/function docstring should be a succinct description of what the method/function does / what it returns

**Review by rly — APPROVED** [review]

**Pull request merged**